catalog/core/catalog-core-api/src/main/java/ddf/catalog/operation/ResultAttributeHighlight.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.operation;
+
+import java.util.List;
+
+/**
+ * ResultAttributeHighlight represents a specific attribute that matched a query and provides
+ * indications of what the specific data the query matched on via a list of highlights. The
+ * individual highlights are meant to be snippets of highlighted data from the attribute's value.
+ */
+public interface ResultAttributeHighlight {
+  String getAttributeName();
+
+  List<String> getHighlights();
```

Comment:
❓ I wonder if instead of returning a string with the Solr implementation detail that the highlights are surrounded with `<em>` if we should return the string without the `<em>` and instead include start and stop position indexes.  Then clients can decide how to represent that data.

Reply:
I agree, returning indices would be cleaner than providing solr text snippets.  The tricky part is pulling the highlights out of the solr response to generate the proper highlight indexes.  However, I think I was able to get this worked out with my latest commits.